### PR TITLE
Enable GET for secret stores to dynamically fetch secrets

### PR DESCRIPTION
**Pull request by cmendible**
# Description

Enabling GET for secret stores to dynamically fetch secrets as described in #1034. Current work just enables the http endpoint.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #1034

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

**Comment by yaron2** [comment]
Great to see this! Can you please resolve conflicts?

**Comment by cmendible** [comment]
@yaron2 solved conflicts in 16d09ff

**Comment by yaron2** [comment]
@cmendible we need the gRPC implementation now. looks good so far.

**Comment by cmendible** [comment]
@yaron2 I have the gRPC code ready but I think I´m missing something on my dev machine cause when I generate the proto code (running ./tools/proto/generate.sh go) it's not complete: both `DaprClient `and `DaprServer `types are not in the resulting `dapr.pb.go`.

Can you tell me how are you generating the files?

**Comment by cmendible** [comment]
> @yaron2 I have the gRPC code ready but I think I´m missing something on my dev machine cause when I generate the proto code (running ./tools/proto/generate.sh go) it's not complete: both `DaprClient `and `DaprServer `types are not in the resulting `dapr.pb.go`.
> 
> Can you tell me how are you generating the files?

Go to generate the files!

File: pkg/http/api.go [hunk 1]
```diff
@@ -350,6 +367,44 @@ func (a *api) onDeleteState(c *routing.Context) error {
 	return nil
 }
 
+func (a *api) onGetSecret(c *routing.Context) error {
+	if a.secretStores == nil || len(a.secretStores) == 0 {
```
Comment (youngbupark) [inline pkg/http/api.go]: would you add unit-test for this ? please check api_test.go

Comment (cmendible) [reply pkg/http/api.go]: @youngbupark, I added some tests in 23ff198

**Comment by yaron2** [comment]
@cmendible are you good to merge this? if so, can you please remove WIP from the title?

**Comment by yaron2** [comment]
We'll need e2e tests for this as this is a new building block, but that should go in a separate PR.

**Comment by cmendible** [comment]
> @cmendible are you good to merge this? if so, can you please remove WIP from the title?

Done!

**Comment by yaron2** [comment]
Requesting final sign off on API from @msfussell

**Pull request merged**